make/compiler/Makefile.gnu

```diff
@@ -180,6 +180,7 @@ endif
 ifeq ($(shell test $(GNU_GPP_MAJOR_VERSION) -eq 8; echo "$$?"),0)
```

Comment:
This is a bug in the way gcc checks for safety.  In the past, this kind of bug has usually been resolved in the next major version of gcc.  It might be helpful to reenable this check someday, so I think it would be better to find out if gcc 9 fails (when that comes out) instead of possibly masking the bug fix by disabling for gcc > 8.

Reply:
Oh, you might be thinking about 8.1, 8.2, ...
We are just checking the major version here, which is 8.

Reply:
I was thinking about 9, but I buy your argument about it likely being fixed in time for 9. 

Has this bug already been reported upstream?

Reply:
There are a number of bugs in the gcc database that might have the same cause.  It is hard to tell, but this bug looks pretty close, and they say it might be a result of complicated control flow.

https://gcc.gnu.org/bugzilla/show_bug.cgi?id=88273